File: clojure/numbers/test/numbers_recursive_test.clj
(ns numbers-recursive-test
  (:require [clojure.test :refer [deftest is run-tests testing]]
            [numbers-recursive :refer [fibonacci factorial sum-numbers]]))

(deftest test-fibonacci
  (testing "Fibonacci recursive function"
    (is (= (fibonacci 5) 5))))

(deftest test-factorial
  (testing "Factorial recursive function"
    (is (= (factorial 5) 120))))

(deftest test-sum-numbers
  (testing "Sum of numbers recursive function"
    (is (= (sum-numbers 5) 15))))

(defn run-all-tests []
  (run-tests 'numbers-recursive-test))

File: dart/numbers/test/numbers_recursive_test.dart
import 'package:test/test.dart';
import '../lib/numbers_recursive.dart';

void main() {
  final numbers = NumbersRecursive();

  group('NumbersRecursive', () {
    test('fibonacci returns correct value for N=10', () {
      expect(numbers.fibonacci(10), equals(55));
    });

    test('factorial returns correct value for N=5', () {
      expect(numbers.factorial(5), equals(120));
    });

    test('sumNumbers returns correct sum for N=5', () {
      expect(numbers.sumNumbers(5), equals(15));
    });
  });
}